### PR TITLE
Added SHA 1 & RIPEMD-160 encoding

### PR DESCRIPTION
**Pull request by unguest**
I've added SHA 1 support since this function still used for file/data integrity in various I.T domains.

### CODE REVIEW

**Comment by unguest** [comment]
RIPEMD-160 is now implemented too 🚀

**Comment by Shankhanil** [comment]
you're on fire. The code will be reviewed soon

File: spec/RIPEMD160.spec.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    describe('Encode/decode testing special characters =>', () => {
+      it('should encode and decode text with spaces', () =>{
+        const clearText = 'I am an HTML developer.';
+        // eslint-disable-next-line max-len
+        const encodeText = 'b623aa45cd96afa80d6b9522e747330ff32e450e';
+        expect(encodeToSHA512(clearText)) == encodeText;
+      });
+      it('Should encode and decode text with special characters', ()=>{
+        // NEED REVIEW
+        const clearText = 'I am $ure buT I am tout à fait sure que c\'est un coup de Fantomas !'; // A very good french reference :p
```
Comment (Shankhanil) [inline spec/RIPEMD160.spec.js:33]: excellent!!

File: spec/SHA1.spec.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+// for tests that need RequireJS
+define(function(require, exports, module) {
+  'use strict';
+  describe('src/SHA512.js =>', ()=>{
+    const encodeToSHA512 = require('../src/convertors/SHA512').encodeToSHA512;
+    it('Should encode and decode pure text', () => {
+      const clearText = 'javascript';
+      // eslint-disable-next-line max-len
+      const encodedText = 'b6e13ad53d8ec41b034c49f131c64e99cf25207a';
+      expect(encodeToSHA512(clearText)) == encodedText;
```
Comment (Shankhanil) [inline spec/SHA1.spec.js:16]: This is the test for SHA1. why have you encoded with SHA512?

File: spec/RIPEMD160.spec.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+ *  File: SHA1.spec.js
+ *  Author(s): unguest (Based on a work of Shankhanil <shankha.rik@gmail.com>)
+ */
+
+require('amd-loader');
+// for tests that need RequireJS
+define(function(require, exports, module) {
+  'use strict';
+  describe('src/SHA512.js =>', ()=>{
+    const encodeToSHA512 = require('../src/convertors/SHA512').encodeToSHA512;
```
Comment (Shankhanil) [inline spec/RIPEMD160.spec.js:11]: RIPEMD160.spec.js encodes using SHA512??? Pls clarify

Comment (unguest) [reply spec/RIPEMD160.spec.js:11]: Oh crap, I didn't used good files (I have several copies of the project...)
I'll fix it today !

**Comment by unguest** [comment]
Here we are, I re-organised my project so I can now avoid errors of this type (pushing from wrong folder with wrong or unfinished files).

File: rebuild.bat [hunk 4]
```diff
@@ -0,0 +1,3 @@
+@echo off
+eslint ./src/*  & "./node_modules/.bin/babel" src -d . & "./node_modules/.bin/jasmine" --random=false & npm install
```
Comment (unguest) [inline rebuild.bat:2]: Reformated the command so it fits windows requirements in terms of syntax.

Comment (Shankhanil) [reply rebuild.bat:2]: Great work on this one. First time contributors would be relieved.

File: spec/SHA1.spec.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    describe('Encode/decode testing special characters =>', () => {
+      it('should encode and decode text with spaces', () =>{
+        const clearText = 'I am an HTML developer.';
+        // eslint-disable-next-line max-len
+        const encodeText = 'd8d2d4f4ccca7ba90a27a751e95a18251a570cef';
+        expect(encodeToSHA1(clearText)) == encodeText;
+      });
+      it('Should encode and decode text with special characters', ()=>{
+        // NEED REVIEW
+        const clearText = 'I am $ure buT I am tout à fait sure que c\'est un coup de Fantomas !'; // A very good french reference :p
```
Comment (Shankhanil) [inline spec/SHA1.spec.js:33]: What review does this line needs?

**Comment by unguest** [comment]
Thanks !

**Pull request merged**